am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,214 @@
+package uk.gov.hmcts.reform.amlib;
+
+import lombok.NonNull;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import uk.gov.hmcts.reform.amlib.enums.AccessType;
+import uk.gov.hmcts.reform.amlib.enums.RoleType;
+import uk.gov.hmcts.reform.amlib.enums.SecurityClassification;
+import uk.gov.hmcts.reform.amlib.exceptions.PersistenceException;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermission;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermissionGrant;
+import uk.gov.hmcts.reform.amlib.models.ResourceAttribute;
+import uk.gov.hmcts.reform.amlib.repositories.DefaultRoleSetupRepository;
+import uk.gov.hmcts.reform.amlib.utils.Permissions;
+
+import javax.sql.DataSource;
+
+public class DefaultRoleSetupImportService {
+    private final Jdbi jdbi;
+
+    public DefaultRoleSetupImportService(String url, String user, String password) {
+        this.jdbi = Jdbi.create(url, user, password);
+        this.jdbi.installPlugin(new SqlObjectPlugin());
+    }
+
+    public DefaultRoleSetupImportService(DataSource dataSource) {
+        this.jdbi = Jdbi.create(dataSource);
+        this.jdbi.installPlugin(new SqlObjectPlugin());
+    }
+
+    /**
+     * Creates a new unique service or updates if already exists.
+     *
+     * @param serviceName the name of the service.
+     */
+    public void addService(@NonNull String serviceName) {
```

Comment:
I'd actually provide only method with 2 args, and `description` would be nullable (and "emptable")

Reply:
Any reason @yarpo to not use method overload here?